internal/uvm/create.go

```diff
@@ -216,9 +273,17 @@ func defaultProcessorCount() int32 {
 }
 
 // normalizeProcessorCount sets `uvm.processorCount` to `Min(requested,
-// runtime.NumCPU())`.
-func (uvm *UtilityVM) normalizeProcessorCount(ctx context.Context, requested int32) {
-	hostCount := int32(runtime.NumCPU())
+// logical CPU count`.
+func (uvm *UtilityVM) normalizeProcessorCount(ctx context.Context, requested int32, processorInfo *hcsschema.ProcessorTopology) {
+	// Use host processor information retrieved from HCS instead of runtime.NumCPU,
+	// GetLogicalProcessorInformationEx or other OS level calls for two reasons.
+	// 1. Go uses GetProcessAffinityMask and falls back to GetSystemInfo both of
+	// which will not return LPs in other NUMA nodes.
+	// 2. GetLogicalProcessorInformationEx will return all processors on the system
+	// but in configurations where the host partition doesn't see the full LP count
+	// i.e "Minroot" scenarios this won't be sufficient.
+	// (https://docs.microsoft.com/en-us/windows-server/virtualization/hyper-v/manage/manage-hyper-v-minroot-2016)
```

Comment:
Why is this a reason to use GetLogicalProcessorInformationEx?

Reply:
This comment is referring to why these other methods (runtime.NumCpu and GetLogicalProcessorInformationEx) are not ideal, not why we're using them.

Reply:
We're getting the processor topology and logical processor count from HCS not these functions. For process isolated we still use runtime.NumCpu which I've mentioned in a comment. These will obviously be run on the host so GetLogicalProcessorInformationEx will be ideal there. I need to add that though

Reply:
Oh, I see. I misread.